tests/unit/test_database.py

```diff
@@ -52,17 +68,31 @@ def test_erase_database(harness_with_postgresql: Harness, monkeypatch: pytest.Mo
         ),
     ]
     cursor_mock.execute.assert_has_calls(calls)
+    harness.cleanup()
 
 
 def test_erase_database_error(
-    harness_with_postgresql: Harness, monkeypatch: pytest.MonkeyPatch
+    datasource_postgresql_password: str, monkeypatch: pytest.MonkeyPatch
 ) -> None:
     """
     arrange: start the Synapse charm, set Synapse container to be ready and set server_name.
     act: add database relation and erase database.
     assert: exception is raised.
     """
-    harness = harness_with_postgresql
+    harness = Harness(SynapseCharm)
+    harness.update_config({"server_name": TEST_SERVER_NAME})
+    relation_id = harness.add_relation("database", "postgresql")
+    harness.add_relation_unit(relation_id, "postgresql/0")
+    harness.update_relation_data(
+        relation_id,
+        "postgresql",
```

Comment:
The config data seems to be the same across the tests, can you extract this somewhere else?

Reply:
I have made the changes here : https://github.com/canonical/synapse-operator/pull/36/commits/c331ecb1a5a485378059e08ebc76026e9c7e7c78